lib/config.js

```diff
@@ -212,127 +87,277 @@ class Config {
          * If user declares "foo", convert to "foo:false".
          */
         this.globals = (options.globals || []).reduce((globals, def) => {
-            const parts = def.split(":");
+            const parts = def.split(SUBCONFIG_SEP);
 
             globals[parts[0]] = (parts.length > 1 && parts[1] === "true");
 
             return globals;
         }, {});
 
-        const useConfig = options.configFile;
-
-        this.options = options;
+        let useConfig = options.configFile;
 
         if (useConfig) {
             debug(`Using command line config ${useConfig}`);
-            if (isResolvable(useConfig) || isResolvable(`eslint-config-${useConfig}`) || useConfig.charAt(0) === "@") {
-                this.useSpecificConfig = loadConfig(useConfig);
-            } else {
-                this.useSpecificConfig = loadConfig(path.resolve(this.options.cwd, useConfig));
+            if (!(isResolvable(useConfig) || isResolvable(`eslint-config-${useConfig}`) || useConfig.charAt(0) === "@")) {
+                useConfig = path.resolve(this.options.cwd, useConfig);
             }
+            this.useSpecificConfig = ConfigFile.loadCached(useConfig);
+        }
+
+        if (this.options.plugins) {
+            Plugins.loadAll(this.options.plugins);
         }
+
+        // Empty values in configs don't merge properly
```

Comment:
I tried allowing empty values in the config and a bunch of tests failed, so I'm pretty sure this is still necessary.